src/map.cpp

```diff
@@ -5652,6 +5656,79 @@ void map::draw( const catacurses::window &w, const tripoint &center )
     }
 }
 
+void map::update_terrain_weather()
+{
+    const w_point weatherPoint = *g->weather_precise;
+    if( g->weather == WEATHER_FLURRIES ) {
+        snowfall += 2;
+    } else if( g->weather == WEATHER_SNOW ) {
+        snowfall += 4;
+    } else if( g->weather == WEATHER_SNOWSTORM ) {
+        snowfall += 8;
+    } else {
+        snowfall -= std::max( 0, ( static_cast<int>( weatherPoint.temperature - 32 ) ) );
+    }
+    snowfall = std::max( 0, snowfall );
+    snowfall = std::min( 172800, snowfall );
+}
+
+void map::add_snow()
+{
+    if( g->weather >= WEATHER_FLURRIES && snowfall > 3000 ) {
+        tripoint gp( 0, 0, 0 );
+        int total_grid = 0;
+        int snowfall_settled = 0;
+        int &gx = gp.x;
+        int &gy = gp.y;
+        int snowfactor = std::min( 15000, snowfall );
+        snowfactor = 15001 - snowfactor;
+        for( gx = 0; gx < my_MAPSIZE * SEEX; ++gx ) {
+            for( gy = 0; gy < my_MAPSIZE * SEEY; ++gy ) {
+                total_grid += 1;
+                const auto grid_offset = point {gx, gy};
+                const tripoint map_location = tripoint( grid_offset, 0 );
+                if( ( !ter( map_location )->has_flag( TFLAG_INDOORS ) ) &&
+                    ( !ter( map_location )->has_flag( "TREE" ) ) && ( is_outside( map_location ) ) &&
+                    ( furn( map_location ) == f_null ) ) {
+                    if( one_in( std::max( 2, static_cast<int>( snowfactor / 2 ) ) ) ) {
```

Comment:
`std::max( 2, static_cast<int>( snowfactor / 2 ) )` can be extracted from the loop as well.